src/ReactPlaceholder.js

```diff
@@ -80,14 +83,28 @@ export default class ReactPlaceholder extends React.Component {
   }
 
   render() {
-    return this.state.ready ? this.props.children : this.getFiller();
+    const { ready, holdPlaceholder } = this.state;
+    return ready && !holdPlaceholder ? this.props.children : this.getFiller();
   }
 
   componentWillReceiveProps(nextProps) {
-    if (!this.props.firstLaunchOnly && this.state.ready && !nextProps.ready) {
-      this.setNotReady();
+    if (!this.props.firstLaunchOnly && !nextProps.ready) {
+      if (this.state.ready) {
+        this.setNotReady();
+      }
+      if (this.props.holdPlaceholder) {
+        clearTimeout(this.holdPlaceholderTimeout);
+        this.setState({ holdPlaceholder: true });
+        this.holdPlaceholderTimeout = setTimeout(() => this.setState({ holdPlaceholder: false }), nextProps.holdPlaceholder);
+      }
     } else if (nextProps.ready) {
       this.setReady();
     }
   }
+
+  componentWillMount() {
+    if (this.props.holdPlaceholder > 0) {
+      this.holdPlaceholderTimeout = setTimeout(() => this.setState({ holdPlaceholder: false }), this.props.holdPlaceholder);
```

Comment:
why are you always setting a timeout? shouldn't it be added only if `this.state.ready` (which is initialized to `this.props.ready`) is `true`?